test/integration/multi_vpc/controls/gcloud.rb

```diff
@@ -0,0 +1,107 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project_id            = attribute('project_id')
+network_01_name       = attribute('network_01_name')
+network_01_routes     = attribute('network_01_routes')
+network_01_route_data = attribute('network_01_route_data')
+network_02_name       = attribute('network_02_name')
+network_02_routes     = attribute('network_02_routes')
+network_02_route_data = attribute('network_02_route_data')
+
+control "gcloud" do
+  title "gcloud configuration"
+
+  network_01_routes.each_with_index do |route, route_index|
```

Comment:
We should avoid dynamically generating tests, especially based on inputs. If a change results in `network_01_routes` being set to an empty list, these tests would not be created and no failure would be reported. This file should be rewritten to be static, relying on attributes only when absolutely necessary.